src/commands/stepfunctions/flare.ts

```diff
@@ -0,0 +1,996 @@
+      this.context.stderr.write(
+        `\nError collecting flare data: ${error instanceof Error ? error.message : String(error)}\n`
+      )
+
+      return 1
+    }
+  }
+
+  /**
+   * Validates required inputs for the flare command
+   * @returns 0 if all inputs are valid, 1 otherwise
+   */
+  private async validateInputs(): Promise<0 | 1> {
+    const errorMessages: string[] = []
+
+    // Validate state machine ARN
+    if (this.stateMachineArn === undefined) {
+      errorMessages.push(helpersRenderer.renderError('No state machine ARN specified. [-s,--state-machine]'))
+    } else {
+      // Validate ARN format
+      const arnPattern = /^arn:aws:states:[a-z0-9-]+:\d{12}:stateMachine:[a-zA-Z0-9-_]+$/
+      if (!arnPattern.test(this.stateMachineArn)) {
+        errorMessages.push(helpersRenderer.renderError('Invalid state machine ARN format.'))
+      }
+    }
+
+    // Validate case ID
+    if (this.caseId === undefined) {
+      errorMessages.push(helpersRenderer.renderError('No case ID specified. [-c,--case-id]'))
+    }
+
+    // Validate email
+    if (this.email === undefined) {
+      errorMessages.push(helpersRenderer.renderError('No email specified. [-e,--email]'))
+    }
+
+    // Validate API key
+    this.apiKey = process.env[CI_API_KEY_ENV_VAR] ?? process.env[API_KEY_ENV_VAR]
+    if (this.apiKey === undefined) {
+      errorMessages.push(
+        helpersRenderer.renderError(
+          'No Datadog API key specified. Set an API key with the DATADOG_API_KEY environment variable.'
+        )
+      )
+    }
+
+    if (errorMessages.length > 0) {
+      for (const message of errorMessages) {
+        this.context.stderr.write(message)
+      }
+
+      return 1
+    }
+
+    return 0
+  }
+
+  /**
+   * Fetches the state machine configuration from AWS
+   * @param sfnClient Step Functions client
+   * @param stateMachineArn ARN of the state machine
+   * @returns State machine configuration
+   */
+  private async getStateMachineConfiguration(
+    sfnClient: SFNClient,
+    stateMachineArn: string
+  ): Promise<DescribeStateMachineCommandOutput> {
+    const command = new DescribeStateMachineCommand({
+      stateMachineArn,
+      includedData: 'ALL_DATA',
+    })
+
+    return sfnClient.send(command)
+  }
+
+  /**
+   * Fetches tags associated with the state machine
+   * @param sfnClient Step Functions client
+   * @param stateMachineArn ARN of the state machine
+   * @returns Map of tag keys to values
+   */
+  private async getStateMachineTags(sfnClient: SFNClient, stateMachineArn: string): Promise<Record<string, string>> {
+    const command = new ListTagsForResourceCommand({
+      resourceArn: stateMachineArn,
+    })
+    const response = await sfnClient.send(command)
+    const tags: Record<string, string> = {}
+    if (response.tags) {
+      for (const tag of response.tags) {
+        if (tag.key && tag.value) {
+          tags[tag.key] = tag.value
+        }
+      }
+    }
+
+    return tags
```

Comment:
could be simplified:
```suggestion
    return Object.fromEntries((response.tags ?? []).filter(({key, value}) => key && value).map(({key, value}) => [key, value]))
```